community/kernel/src/test/java/org/neo4j/graphdb/impl/notification/NotificationCodeTest.java

```diff
@@ -45,4 +49,24 @@ public void shouldConstructNotificationFor_INDEX_HINT_UNFULFILLABLE()
         assertThat( notification.getDescription(), equalTo( "The hinted index does not exist, please check the schema (hinted index is index on :Person(name))" ) );
     }
 
+    @Test
+    public void shouldConstructNotificationFor_CARTESIAN_PRODUCT()
+    {
+        Map<String, InputPosition> map = new HashMap<>( );
+        map.put( "n", new InputPosition( 0, 1, 3 ) );
+        map.put( "node2", new InputPosition( 2, 3, 4 ));
+        NotificationDetail identifierDetail = NotificationDetail.Factory.cartesianProduct( map );
+        Notification notification = CARTESIAN_PRODUCT.notification( InputPosition.empty, identifierDetail );
+
+        assertThat( notification.getTitle(), equalTo( "This query builds a cartesian product between disconnected patterns." ) );
+        assertThat( notification.getSeverity(), equalTo( SeverityLevel.WARNING ) );
+        assertThat( notification.getCode(), equalTo( "Neo.ClientNotification.Statement.CartesianProduct" ) );
+        assertThat( notification.getPosition(), equalTo( InputPosition.empty ) );
+        assertThat( notification.getDescription(), equalTo( "If a part of a query contains multiple disconnected patterns, this will build a cartesian product " +
+                                                            "between all those parts. This may produce a large amount of data and slow down query processing. While " +
+                                                            "occasionally intended, it may often be possible to reformulate the query that avoids the use of this cross " +
+                                                            "product, perhaps by adding a relationship between the different parts or by using OPTIONAL MATCH " +
+                                                            "(identifiers are: n at (0, 1, 3), node2 at (2, 3, 4))" ) );
+    }
```

Comment:
I find that printing the input positions like that is not helpful, actually I'm not sure that including them in the details string is helpful at all. Why not rather include the sets of connected identifiers? Something like: "(first connected component is n, a, second connected component is m, b)"